sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -115,14 +115,20 @@ function buildPipelineRequest(
   url: string,
   options: InternalRequestParameters = {}
 ): PipelineRequest {
-  const { body, formData } = getRequestBody(options.body, options.contentType);
+  const { body, formData } = getRequestBody(
+    options.body,
+    options.contentType ?? (options.headers?.["content-type"] as string)
```

Comment:
Is `options.headers?.["content-type"]` always a string? I think it can also be undefined, right?

Reply:
There is also a question about precedence, should we error if both `options.contentType` and the header are set?

Reply:
Actually, it can also be a number and boolean, as we defined the header like https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-client-rest/src/sendRequest.ts#L94-L107 

I have no preference about the precedence but not sure if we should error, feel like error should be the last option here ? @xirzec @joheredi how do you think ?

Reply:
I usually lean towards the manually provided option having precedence if I understand the problem correctly. 

Perhaps we could log a warning if we are worried about the two being different?

Reply:
resolve it as offline confirmed that we don't want to add azure logger as the dependency.